fe/fe-core/src/main/java/org/apache/doris/plugin/dialect/HttpDialectUtils.java

```diff
@@ -30,18 +31,90 @@
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
  * This class is used to convert sql with different dialects using sql convertor service.
  * The sql convertor service is a http service which is used to convert sql.
+ * <p>
+ * Features:
+ * - Support multiple URLs (comma separated)
+ * - Blacklist mechanism for failed URLs
+ * - Automatic failover and retry
+ * - URL caching and smart selection
  */
 public class HttpDialectUtils {
     private static final Logger LOG = LogManager.getLogger(HttpDialectUtils.class);
 
-    public static String convertSql(String targetURL, String originStmt, String dialect,
+    // Cache URL manager instances to avoid duplicate parsing
+    private static final ConcurrentHashMap<String, UrlManager> urlManagerCache = new ConcurrentHashMap<>();
```

Comment:
We need to consider how to remove the unused entries.
So I suggest to use a Caffein Cache, and set a expire time so that it can remote unused entries automatically.